storage/market.go

```diff
@@ -43,15 +43,28 @@ func (s *Storage) GetTicker(coin, token string) (blockatlas.Ticker, error) {
 	return *cd, nil
 }
 
-func (s *Storage) SaveRates(rates blockatlas.Rates) {
+type RateProviderList interface {
+	GetPriority(providerId string) int
+}
+
+func (s *Storage) SaveRates(rates blockatlas.Rates, pl RateProviderList) {
 	for _, rate := range rates {
 		r, err := s.GetRate(rate.Currency)
-		if err == nil && rate.Timestamp < r.Timestamp {
-			return
+		if err == nil {
```

Comment:
Does it make sense to move this condition to separate func since it's been used twice?

Reply:
the objects are different, I tried to do that, but we need to create some interfaces and the code starts to be bigger